gr-digital/python/digital/qa_constellation.py

```diff
@@ -177,6 +178,26 @@ def setUp(self):
     def tearDown(self):
         pass
 
+    def test_normalization(self):
+        rot_sym = 1
+        side = 2
+        width = 2
+        # Test a couple of basic constellations
+        for constel_points, code in (digital.psk_4_0(), digital.qam_16_0()):
+            constel = digital.constellation_rect(constel_points, code, rot_sym,
+                                                 side, side, width, width,
+                                                 constellation.POWER_NORMALIZATION)
+
+            points = np.array(constel.points())
+            avg_power = np.sum(abs(points)**2) / len(points)
+            self.assertAlmostEqual(avg_power, 1.0, 6)
+            constel = digital.constellation_rect(constel_points, code, rot_sym,
+                                                 side, side, width, width,
+                                                 constellation.AMPLITUDE_NORMALIZATION)
+            points = np.array(constel.points())
+            avg_amp = np.sum(abs(points)) / len(points)
+            self.assertAlmostEqual(avg_amp, 1.0, 6)
```

Comment:
I prefer
```suggestion
            self.assertAlmostEqual(avg_amp, 1.0, places=6)
```
...because that method has so many ways of defining "almost". But not a hold-up. This is just my preference.